### PR TITLE
CBG-2506: Tuning import partitions for when in serverless mode

### PR DESCRIPTION
**Pull request by gregns1**
CBG-2506

Small PR to tune the import partitions for when running sync gateway in serverless mode. Also added a unit test to test the functionality with one case proving the default is set in serverless mode and another proving the value is not overwritten when import partitions are specified in the config. 

## Pre-review checklist
- [x] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [x] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [x] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1102/

### CODE REVIEW

File: rest/server_context.go [hunk 1]
```diff
@@ -655,7 +655,9 @@ func dbcOptionsFromConfig(ctx context.Context, sc *ServerContext, config *DbConf
 	}
 	importOptions.BackupOldRev = base.BoolDefault(config.ImportBackupOldRev, false)
 
-	if config.ImportPartitions == nil {
+	if config.ImportPartitions == nil && sc.Config.IsServerless() {
```
Comment (adamcfraser) [inline rest/server_context.go]: It's a bit of a style nit, but I think I would implement a `DefaultImportPartitions(serverless bool) uint16` function in base to encapsulate the handling here and above, particularly given the complexity it's adding to this particular if block.

**Review by adamcfraser — CHANGES_REQUESTED** [review]
Minor style comment, otherwise looks good.

File: rest/serverless_test.go [hunk 2]
```diff
@@ -579,3 +579,43 @@ func TestServerlessUnsuspendAdminAuth(t *testing.T) {
...
+	}{
+		{
+			name:               "serverless partitions",
+			expectedPartitions: base.Uint16Ptr(6),
+		},
+		{
+			name:               "serverless partitions with import_partition specified",
+			importPartition:    base.Uint16Ptr(8),
+			expectedPartitions: base.Uint16Ptr(8),
+		},
```
Comment (bbrks) [inline rest/serverless_test.go]: Is it worth adding the test case for `serverless: false` == `16` or do we have sufficient coverage elsewhere?

**Review by bbrks — DISMISSED** [review]
LGTM but see what you think about the comment

File: rest/serverless_test.go (hunk 2 above)
Comment (gregns1) [reply rest/serverless_test.go]: Good point! Updated the PR to add this test case in.

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**